src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] dataElement;
```

Comment:
The variable name `dataElement` is not very informative. Consider renaming it to something like `elements` or `dataElements` to better reflect that it holds multiple elements.